account-kit/react/src/components/button.tsx

```diff
@@ -34,44 +37,43 @@ export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
       }
     })();
 
+    const [hideChildren, setHideChildren] = useState(false);
+
+    useLayoutEffect(() => {
+      if (!localRef) return;
+
+      const parent = localRef.parentElement;
+      if (!parent) return;
+      const siblings = parent.children;
+
+      // this assumes this element is in a button group
+      // in a 3 element button group all buttons have their text shown
+      if (siblings.length <= 3) return;
```

Comment:
Sorry a little confused what is the thing we are trying to achieve with the UI. To me this feels a little in-direct. If a child view is reaching out to understand parent context that feels like a code smell. Especially in a broad component like button.

Reply:
I agree generally about not relying on your parents for styling your self, but here this was probably the easiest way to achieve this layout:
<img width="683" alt="image" src="https://github.com/user-attachments/assets/0b2ccd6c-b6a6-4654-9169-de5997464aae">

There's a couple wrinkles:
1. since a button group is max 2 rows, I did this by having a flex wrap + setting the first child at max width and the rest of them having a flex grow and flex shrink. This is done on the `btn-group` class. Because the children are flex grow and shrink and the container is flex wrap, the buttons will shrink only until the text is not cut off. So we can't do something in the button like `if (width < text.width + icon.width) hideText` because that will never be the case and the buttons will overflow into the next row.
2. If we made a button group component, the component itself wouldn't be able to hold this logic because it would just receive children and AFAIK it's not straightforward to check what kind of child it is and pass in new props
3. the button group is just the high level container for each auth section. Almost all of the sections are buttons so that's chill, but you can also have email as a section. The email section is typically rendered in the first section and someone could add other sections along side email. In that case, you would want the remaining sections (which are all buttons) to render as the second row of a button group.

This ultimately felt like the easiest thing to do to match the layout. Buttons aren't actually exported to be consumed externally, so this is all internal to the SDK which feels like an ok place to make these kinds of one offs

Reply:
Hm, this is pretty tricky. So if I'm understanding correctly, the goal of this code is to apply different styling if the button has more than three siblings (counting itself)?

I'm okay with this solution, but did you look into using tricky CSS selectors to apply based on sibling count? I think it might be possible to do something like
```css
.parent:has(> :nth-child(4)) > * {
    /* ... */
}
```
where that rule would match elements that are a child of a `.parent` element if that `.parent` has at least four children. If that works, it might be a little simpler.

Reply:
oh that's cool, lemme try that

Reply:
ok sweet! check it out now? I moved it into css, though I think the smell still exists here because it does require specific set up between btn and btn groups, but I think this is ok. At somepoint, we will need to support the case where there are more than 4 buttons in the second row. What we will have to do there is: if there are 4 buttons, they get condensed into a `...` button that opens some other view containing the overflow. To handle that we'll have to use a `ref` that checks either sibling counts or children, so we might not be out of the woods fully. That said I like the css solution more because it keeps all of this btn and btn group styling in the same place, so it's a bit more discoverable

Reply:
That sort of thing may still be possible with CSS if the "..." button is always in the DOM but hidden by CSS unless there are more than four buttons in the second row. But we can cross that bridge when we come to it. The new version looks good!